### PR TITLE
Forget cached account

### PR DESCRIPTION
**Pull request by oldalton**
## Proposed changes

In order to make sure user has a way to "forget" or "wipe" their account entry in cache, we need to add new API to MSAL to do so. Additionally, when SSO extension is present, MSAL needs to propagate the signal to the SSO extension (will only be allowed for privileged/whitelisted apps like broker host apps). This PR adds "wipe account" flag to the SSO extension signout request.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by oldalton** [comment]
MSAL and broker PR check will be fixed after PRs in those repositories are fixed.

File: IdentityCore/src/controllers/broker/MSIDSSOExtensionSignoutController.h [hunk 1]
```diff
@@ -27,9 +27,20 @@
...
 API_AVAILABLE(ios(13.0), macos(10.15))
 @interface MSIDSSOExtensionSignoutController : MSIDSignoutController
 
+@property (nonatomic, readonly) BOOL shouldWipeAccount;
+
+- (instancetype)initWithRequestParameters:(MSIDInteractiveRequestParameters *)parameters
+                 shouldSignoutFromBrowser:(BOOL)shouldSignoutFromBrowser
+                        shouldWipeAccount:(BOOL)shouldWipeAccount
+                             oauthFactory:(MSIDOauth2Factory *)oauthFactory
+                                    error:(NSError * _Nullable * _Nullable)error;
```
Comment (NerevarineRule) [inline IdentityCore/src/controllers/broker/MSIDSSOExtensionSignoutController.h:42]: nit: should we remove space between * and _ ?

**Review by NerevarineRule — APPROVED** [review]

File: IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationSignoutFromDeviceRequest.m [hunk 2]
```diff
@@ -94,8 +96,9 @@ - (NSDictionary *)jsonDictionary
     }
     
     json[MSID_PROVIDER_TYPE_JSON_KEY] = MSIDProviderTypeToString(self.providerType);
-    json[MSID_SIGNOUT_FROM_BROWSER_KEY] = @(_signoutFromBrowser);
-    json[MSID_CLEAR_SSO_EXT_COOKIES_KEY] = @(_clearSSOExtensionCookies);
+    json[MSID_SIGNOUT_FROM_BROWSER_KEY] = [NSString stringWithFormat:@"%d", (int)_signoutFromBrowser];
```
Comment (jasoncoolmax) [inline IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationSignoutFromDeviceRequest.m:99]: will this cause compatibility issue?

Comment (jasoncoolmax) [reply IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationSignoutFromDeviceRequest.m:99]: ignore my comment, our code can support both NSString and NSNumber :)

**Review by jasoncoolmax — APPROVED** [review]

File: IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationSignoutFromDeviceRequest.m (hunk 2 above)
Comment (oldalton) [reply IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationSignoutFromDeviceRequest.m:99]: NSNumber actually doesn't even get sent when converted to query items :) So this was essentially broken and never sent in current version.

**Pull request merged**